new_experiment/nbconverted/viz_expression_data.py

```diff
@@ -0,0 +1,482 @@
+
+
+# Normalize simulated_data
+# Load pickled file
+with open(scaler_filename, "rb") as scaler_fh:
+    scaler = pickle.load(scaler_fh)
+
+normalized_simulated_data = scaler.transform(simulated_data)
+
+normalized_simulated_data = pd.DataFrame(
+    normalized_simulated_data,
+    columns=simulated_data.columns,
+    index=simulated_data.index,
+)
+
+print(normalized_simulated_data.shape)
+normalized_simulated_data.head()
+
+
+# In[17]:
+
+
+"""# If template experiment included in training compendium
+# Get normalized template data
+sample_ids = list(template_data.index)
+normalized_template_data = normalized_compendium_data.loc[sample_ids]
+
+print(normalized_template_data.shape)
+normalized_template_data.head()"""
+
+
+# In[18]:
+
+
+# If template experiment NOT included in training compendium
+with open(scaler_filename, "rb") as scaler_fh:
+    scaler = pickle.load(scaler_fh)
+
+normalized_template_data = scaler.transform(template_data)
+
+normalized_template_data = pd.DataFrame(
+    normalized_template_data,
+    columns=template_data.columns,
+    index=template_data.index,
+)
+
+
+# In[19]:
+
+
+# Label samples 
+normalized_compendium_data['sample group'] = "compendium"
+normalized_template_data['sample group'] = "template"
+normalized_simulated_data['sample group'] = "simulated"
+
+
+# In[20]:
+
+
+normalized_all_data = pd.concat([normalized_template_data,
+                             normalized_simulated_data,
+                             normalized_compendium_data
+])
+
+
+# In[21]:
+
+
+# Plot
+
+# Drop label column
+normalized_all_data_numeric = normalized_all_data.drop(['sample group'], axis=1)
+
+model = umap.UMAP(random_state=1).fit(normalized_all_data_numeric)
+
+normalized_all_data_UMAPencoded = model.transform(normalized_all_data_numeric)
+normalized_all_data_UMAPencoded_df = pd.DataFrame(data=normalized_all_data_UMAPencoded,
+                                         index=normalized_all_data.index,
+                                         columns=['1','2'])
+
+# Add back label column
+normalized_all_data_UMAPencoded_df['sample group'] = normalized_all_data['sample group']
+
+# Plot
+fig = ggplot(normalized_all_data_UMAPencoded_df, aes(x='1', y='2'))
+fig += geom_point(aes(color='sample group'), alpha=0.1)
+fig += labs(x ='UMAP 1',
+            y = 'UMAP 2',
+            title = 'Gene expression data in gene space')
+fig += theme_bw()
+fig += theme(
+    legend_title_align = "center",
+    plot_background=element_rect(fill='white'),
+    legend_key=element_rect(fill='white', colour='white'), 
+    legend_title=element_text(family='sans-serif', size=15),
+    legend_text=element_text(family='sans-serif', size=12),
+    plot_title=element_text(family='sans-serif', size=15),
+    axis_text=element_text(family='sans-serif', size=12),
+    axis_title=element_text(family='sans-serif', size=15)
+    )
+fig += scale_color_manual(['#bdbdbd', 'red', 'blue'])
+fig += guides(colour=guide_legend(override_aes={'alpha': 1}))
+
+print(fig)
```

Comment:
The plots serve two purposes: 1) I wanted to confirm that the latent space shifting looked as expected (i.e. the structure of the experiments were preserved). This was to rule out a mistake in the simulation that might be the reason for the volcano plots looking not as expected. Since the shift is linear that is the reason I was using PCA. But I have also added UMAP to look at the latent space as a whole in addition to the first two PCs. 2) I wanted to confirm that the simulated experiments were within the bounds of the data as opposed to a valley that might explain the consistency in patterns.

Reply:
Got it! That makes sense.